nemo_curator/modules/fuzzy_dedup.py

```diff
@@ -1691,7 +1700,12 @@ def _batched_merge_and_write(
 
             subset_ddf = subset_ddf[[self.left_id, self.right_id, "jaccard"]]
             output_batch_path = os.path.join(output_path, f"{batch_id}.parquet")
-            subset_ddf.to_parquet(output_batch_path, write_index=False)
+            if batch_id == 0:
+                subset_ddf.to_parquet(
+                    output_batch_path, write_index=False, overwrite=True
+                )
+            else:
+                subset_ddf.to_parquet(output_batch_path, write_index=False, append=True)
```

Comment:
Can the reviewer TAL at this closely

Reply:
Thinking out loud: If a prev job wrote `encoded_jaccard_pair` with `batch_0.parquet`, `batch_1.parquet` ... `batch_n.parquet`. In theory the script would need to overwrite all batches rather than append for the batch_id != 0 case. Cases like these can however be problematic:  Is a prev job wrote more batches than the current one is trying to write we only overwrite a few batch files but don't touch the extra batches written by a previous job. In this case removing the whole directory where `encoded_jaccard_pair` will be written is cleaner wdyt?

Reply:
Actually now that I've looked at #302 this codepath is going away, so ignore my comments.

Reply:
Good catch @sarahyurick, looking at the code, `overwrite` only removes when path is directory (it has `isdir` check) https://github.com/dask/dask/blob/7ab0c93cfca0af130c778f9c1baec29a1f10b616/dask/dataframe/io/parquet/core.py#L860

I guessed it does for filepaths too, but that doesn't make sense, now that I think of it. I saw a similar pattern in lsh ([lines here](https://github.com/NVIDIA/NeMo-Curator/blob/dc727828bed491933fddfa166cc4c5ad865cfba7/nemo_curator/modules/fuzzy_dedup.py#L358-L365)) from where I was _inspired_, but the path there seems to be directory path (tho I'm intrigued how are filenames decided)

Any how looks like after the merge this doesn't need to be addressed but thanks for catching this